MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -444,22 +449,31 @@ public string GetAudioEncoder(EncodingJobInfo state)
         public string GetInputArgument(EncodingJobInfo state, EncodingOptions encodingOptions)
         {
             var arg = new StringBuilder();
+            var videoDecoder = GetHardwareAcceleratedVideoDecoder(state, encodingOptions);
+            var outputVideoCodec = GetVideoEncoder(state, encodingOptions);
 
-            if (state.IsVideoRequest
-                && string.Equals(encodingOptions.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase))
+            if (state.IsVideoRequest && string.Equals(encodingOptions.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase))
             {
-                arg.Append("-hwaccel vaapi -hwaccel_output_format vaapi")
-                    .Append(" -vaapi_device ")
-                    .Append(encodingOptions.VaapiDevice)
-                    .Append(' ');
+                // While using VAAPI decoder
+                if ((videoDecoder ?? string.Empty).IndexOf("vaapi", StringComparison.OrdinalIgnoreCase) != -1)
```

Comment:
I'm starting to grow tired of seeing this `.IndexOf("foo", StringComparison.OrdinalIgnoreCase) != -1` thing... maybe we should introduce a struct of `bool`s with fields like `isVaapiDecoder`, `isQsvEncoder`, etc., fill it once in the beginning and then use proper field names? It would make comments unnecessary because the code itself would speak.